feature-flags-tmp/feature_flags/feature_flags.py

```diff
@@ -0,0 +1,222 @@
+import copy
+import re
+
+from fastapi import HTTPException
+from typing import Optional
+
+import reflex as rx
+from sqlmodel import Field
+
+
+class FeatureFlags(rx.Model, table=True):
+    name: str = Field(primary_key=True)
+    value: str = Field()
+
+
+class FeatureFlagsState(rx.State):
+    feature_flags_as_loaded_from_db: Optional[dict[str, str]] = None
+
+    pending_creates_or_updates: dict[str, str] = {}
+    pending_deletes: set[str] = set()
+
+    new_flag_modal_is_open: bool = False
+    new_flag_modal_error: Optional[str] = None
+
+    new_flag_modal_flag_name: Optional[str] = ""
+    new_flag_modal_flag_value: Optional[str] = ""
+
+    @rx.var
+    def save_button_color(self) -> str:
+        if self.pending_creates or self.pending_deletes or self.pending_updates:
+            return "red"
+        else:
+            return ""
+
+    def load_feature_flags_from_db(self):
+        """Populates self.feature_flags_as_loaded_from_db"""
+        with rx.session() as session:
+            feature_flags = session.exec(FeatureFlags.select).all()
+        self.feature_flags_as_loaded_from_db = {ff.name: ff.value for ff in feature_flags}
+
+    @rx.var
+    def feature_flag_name_value_pairs(self) -> list[tuple[str, str]]:
+        if self.feature_flags_as_loaded_from_db is None:
+            self.load_feature_flags_from_db()
```

Comment:
side effects not advised in computed var

Reply:
The intent is a "cached property".  I.e. load stuff from DB if never loaded before.  What would be a better pattern?

E.g. is there logic I can attach to "initialization" of the state?